File: ch3_vector/point2line/point2line.cpp
#include "point2line.h"
#include <maya/MPoint.h>
#include <maya/MVector.h>
#include <maya/MGlobal.h>

MStatus Point2Line::doIt(const MArgList & args)
{
	MPoint p(1.0, 3.0, 4.0);
	// line
	MPoint p0(0, 0, 0);
	MPoint p1(1, 0, 0);

	MVector a = p - p0;
	MVector b = p1 - p0;
	b.normalize();

	MVector c = (a * b) * b;
	MVector d = a - c;
	double dist = d.length();
	MGlobal::displayInfo(MString("distance: ") + dist);

	return MS::kSuccess;
}

void * Point2Line::creator()
{
	return new Point2Line();
}

File: ch3_vector/point2line.py
# coding:utf-8
# calculate vector(point) perpendicular to line(pointB-pointA)
# trans from C++

from maya import OpenMaya as om

p = om.MPoint(1,3,4)
p0 = om.MPoint(0,0,0)
p1 = om.MPoint(1,0,0)

a = p - p0
type(a)
type(p)
b = p1 - p0
b.normalize()
#c = (a * b)*b ##ERROR
c = b * (a * b)
d = a - c
om.MGlobal.displayInfo('distance: %s'%d.length())
